### ISSUE TITLE
Enc-Dec not learning

### ISSUE THREAD

**Issue by MemduhG** [lucidrains/reformer-pytorch#125]
Hi, I am trying to use the EncDec to train a translation task, but experimenting with it on just one batch, the loss does not seem to go down. I feel like I should be doing something different with masking, but it looks like causal is automatically set to true in the Decoder. Should I compute some mask myself for the tgt input to the decoder, and pass it explicitly myself?

My BOS index is 1 and the EOS is 2. I don't know if I should be doing any shifting of the decoder input, but it looks like you handle that internally.

```
from reformer_pytorch import ReformerEncDec

DE_SEQ_LEN = 1024
EN_SEQ_LEN = 1024

enc_dec = ReformerEncDec(
    dim = 512,
    enc_num_tokens = 32000,
    enc_depth = 6,
    enc_max_seq_len = DE_SEQ_LEN,
    dec_num_tokens = 32000,
    dec_depth = 6,
    dec_max_seq_len = EN_SEQ_LEN
).cuda()

train_seq_in = torch.transpose(nn.utils.rnn.pad_sequence(src_raw, padding_value=3), 0, 1).cuda() # 10, 119
train_seq_out = torch.transpose(nn.utils.rnn.pad_sequence(tgt_raw, padding_value=3), 0, 1).cuda() # 10, 441
input_mask = train_seq_in != 3

for i in range(1000):
    loss = enc_dec(train_seq_in, train_seq_out, return_loss = True, enc_input_mask = input_mask)
    print(loss)
    loss.backward()
```

The `src_raw` and `tgt_raw` are just lists of one-dimensional long tensors that I make a batch out of.

The loss seems to just stay around 10.4. What could I be doing wrong?

**Comment by MemduhG** [comment]
I wasn't stepping with an optimizer, embarrassing 😄 I'll get the hang of pytorch some day.
